content/program-testing.md

```diff
@@ -0,0 +1,534 @@
+            program_pack::Pack,
+            system_program::ID as SYSTEM_PROGRAM_ID,
+        },
+        solana_program_test::*,
+        solana_sdk::{
+            signature::Signer, signer::keypair::Keypair,
+            system_instruction::create_account, transaction::Transaction,
+        },
+        spl_associated_token_account::{
+            get_associated_token_address,
+            instruction::create_associated_token_account,
+        },
+        spl_token::{instruction::initialize_mint, state::Mint, ID as TOKEN_PROGRAM_ID},
+    };
+}
+```
+
+Next, we’ll declare our first unit test and initialize the testing environment.
+
+```rust
+#[cfg(test)]
+mod tests {
+    use {
+        super::*,
+        assert_matches::*,
+        solana_program::{
+            instruction::{AccountMeta, Instruction},
+            system_program
+        },
+        solana_program_test::*,
+        solana_sdk::{signature::Signer, transaction::Transaction},
+        spl_token::*,
+    };
+
+    // First unit test
+    #[tokio::test]
+    async fn it_works() {
+        let program_id = Pubkey::new_unique();
+        let (mut banks_client, payer, recent_blockhash) = ProgramTest::new(
+            "bpf_program_template",
+            program_id,
+            processor!(process_instruction),
+        )
+        .start()
+        .await;
+
+        ...
+    }
+}
+```
+
+### 3. Construct transaction
+
+Our first unit test will test whether or not our program is actually functioning as intended, to test this we’ll need to create a transaction to submit to the program. We'll be making use of the `solana_sdk` crate to help us do this, it may help to think about the steps you would need to go through to build this transaction from a typescript client when doing this.
+
+The code in the remainder of the demo should go inside the `it_works()` testing function we just created.
+
+First, we are going to create a token mint and assign the mint authority to a PDA of the Movie Review program so that it has the capability of minting tokens to users.
+
+```rust
+// Derive PDA for token mint authority
+let (mint_auth, _bump_seed) = Pubkey::find_program_address(&[b"tokens"], &program_id);
+
+// Create mint account
+let mint_keypair = Keypair::new();
+let rent = banks_client.get_rent().await.unwrap();
+let mint_rent = rent.minimum_balance(Mint::LEN);
+let create_mint_acct_ix = create_account(
+    &payer.pubkey(),
+    &mint_keypair.pubkey(),
+    mint_rent,
+    Mint::LEN.try_into().unwrap(),
+    &TOKEN_PROGRAM_ID,
+);
+// Create initialize mint instruction
+let init_mint_ix = initialize_mint(
+    &TOKEN_PROGRAM_ID,
+    &mint_keypair.pubkey(),
+    &mint_auth,
+    Some(&mint_auth),
+    9,
+)
+.unwrap();
+```
+
+We use two functions that we've imported from Solana crates to help create the `create_account` and `initialize_mint` instructions.
+
+Next, we need to create/derive the review, comment counter, and user associated token account addresses.
```

Comment:
@ixmorrow suggestion: 
from: Next, we need to create/derive the review,  ...

to: Next, we need to derive the review, ...

...mostly I just dislike the undecisive verb slash. Maybe just "Next, we need to find the review, ..."?